Lacework/lacework_module/lacework_connector.py

```diff
@@ -121,55 +138,58 @@ def get_next_events(self, cursor: str | None) -> Any | None:
     def _get_most_recent_timestamp_from_items(self, items: list[dict[Any, Any]]) -> float:
         def _extract_timestamp(item: dict[Any, Any]) -> float:
             RFC3339_STRICT_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
-            return datetime.datetime.strptime(item["startTime"], RFC3339_STRICT_FORMAT).timestamp()
+            return datetime.strptime(item["startTime"], RFC3339_STRICT_FORMAT).timestamp()
 
         later = 0.0
         for item in items:
             if _extract_timestamp(item) > later:
                 later = _extract_timestamp(item)
-        return later
+        return float(later)
 
     def forward_next_batches(self) -> None:
         """
         Successively queries the Lacework Central API while more are available
         and the current batch is not too big.
         """
-        has_more_messages = True
-        cursor = self.cursor
-
-        messages = []
-        while has_more_messages and self.running:
-            has_more_messages = False
-
-            batch = self.get_next_events(cursor)
-            if batch is None:
-                break
+        now = datetime.now(timezone.utc)
 
-            if batch.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage") != None:
-                has_more_messages = True
-                cursor = max(item.get("startTime") for item in batch.get("data", []))
+        first_batch = self.get_response_by_timestamp()
+        if first_batch:
+            next_page_url = first_batch.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
+            items = first_batch.get("data", [])
 
-            items = batch.get("data", [])
-            if len(items) > 0:
-                most_recent_timestamp_seen = self._get_most_recent_timestamp_from_items(items)
-                events_lag = int(time.time() - most_recent_timestamp_seen)
-                EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
-                INCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(items))
+            last_date = self._get_most_recent_timestamp_from_items(items)
+            events_lag = int(time.time() - last_date)
+            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
```

Comment:
I suggest saving the most recent date seen here for the first batch
```suggestion
            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
            self.most_recent_date_seen = last_date
```